Notebooks/word2vec/tf_Word2Vec.py

```diff
@@ -0,0 +1,120 @@
+# *******************************************************************************
+# **
+# **  Author: Michael Lomnitz (mrlomnitz@lbl.gov)                               
+# **  Python module defining embeddigns (word2vec) model using skipgram or 
+# **  continuous bag or words to define word context. 
+# **
+# *******************************************************************************
+# Modules
+from __future__ import print_function
+import math
+import numpy as np
+import random
+import tensorflow as tf
+import collections
+from matplotlib import pylab
+from six.moves import range
+from itertools import compress
+
+# Some model constants
+data_index = 0
+embedding_size = 128 # Dimension of the embedding vector.
+num_sampled = 64 # Number of negative examples to sample.
+
+# Generate the batch and associated labels for training in skipgram
+def generate_batch(batch_size, num_skips, skip_window, data):
+    global data_index
```

Comment:
globals are considered a bit heavy handed in python, might be good to explain why you explicitly make this a global